### ISSUE TITLE
Spotless is leaking Java 11 dependency.

### ISSUE THREAD

**Issue by JavierSegoviaCordoba** [diffplug/spotless#1032]
## Summary
I have a published plugin that is configuring Spotless and I apply it in all my projects, after upgrading Spotless to 6.0.3, My project, which is Java 8, which is applying my published plugin can't be compiled because of a dependency is Java 11.

```
The consumer was configured to find a runtime of a library compatible with Java 11, 
packaged as a jar, preferably optimized for standard JVMs, and its dependencies 
declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with 
value 'jvm'. However we cannot choose between the following variants of project :mokoki-core:
    - jvmRuntimeElements
    - spotless1316845943
```

`spotless1316845943` this should not be appearing there I think, but it is being leaked into the project.

I am not having this issue with 6.0.2, but 6.0.4 has this issue too, not only 6.0.3.

- Gradle 7.3.1
- Windows 10

My published plugin is adding Spotless as a dependency (`implementation(...)`). So the consumer hasn't to add Spotless to the classpath itself, because the plugin has the dependency directly.

**Comment by nedtwigg** [comment]
Related to #1027

**Comment by nedtwigg** [comment]
Should be fixed in `6.1.0`, please reopen if the issue persists.  See [these docs](https://github.com/diffplug/spotless/tree/main/plugin-gradle#dependency-resolution-modes) for more details.
